libs/frontend/util-edit-text/src/lib/prosemirror-utils.ts

```diff
@@ -48,7 +49,19 @@ export const firstMarkInSelection = (
   }
   const { $cursor, ranges } = selection
   if ($cursor) {
-    return mark.isInSet(state.storedMarks || $cursor.marks())
+    const marks = $cursor.marks()
+    const markInSet = mark.isInSet(state.storedMarks || marks)
+    if (markInSet) {
+      return markInSet
+    } else {
+      // FIXME: this is a workaround for link mark not being found in `mark.isInSet`.
+      const firstMark = marks[0]
```

Comment:
I'm not sure why `firstMarkInSelection` always returns false for link mark 🤔.
So I made a temporary workaround that check `mark.type.name`.

Reply:
I see! I'll try to remember to investigate this, but probably won't have time for a while. It takes me a while to solve anything ProseMirror related, I feel like I have to relearn things every time I look at it 😅